contractcourt/briefcase.go

```diff
@@ -1184,6 +1201,94 @@ func (b *boltArbitratorLog) checkpointContract(c ContractResolver,
 	}, func() {})
 }
 
+// InsertCanceledHTLCs inserts the given set of canceled HTLCs into the
+// database. It converts the set to a slice and persists the slice.
+//
+// NOTE: Part of the ArbitratorLog interface.
+func (b *boltArbitratorLog) InsertCanceledHTLCs(
```

Comment:
I think these commits can be move to a new PR where we want to cancel non-dust incoming HTLCs.

Reply:
hmm not sure, it solves the case for outgoing_htlc_resolver, when we restart before the second stage htlc is confirmed ?